packages/linter/src/rules/AmpImgUsesSrcSet.ts

```diff
@@ -0,0 +1,49 @@
+import { Context, Result } from "../index";
+import { Rule } from "../rule";
+import { notPass } from '../filter';
+
+const CHECKED_IMG_LAYOUTS = ["fill", "flex-item", "intrinsic", "responsive"];
+
+const SVG_URL_PATTERN = /^[^?]+\.svg(\?.*)?$/i
+
+export class AmpImgUsesSrcSet extends Rule {
+  async run(context: Context) {
+    const $ = context.$;
+
+    const incorrectImages = $("amp-img")
+        .filter((_, e) => {
+          const src = $(e).attr("src");
+          let layout = $(e).attr("layout");
+          const srcset = $(e).attr("srcset");
+          const parent = $(e).parent();
+          if (parent.prop("tagName").startsWith('AMP-')) {
+            const parentLayout = $(parent).attr("layout");
+            if (parentLayout) {
+              layout = parentLayout;
+            }
+          }
+          return (
+            !SVG_URL_PATTERN.exec(src)
+            && layout && CHECKED_IMG_LAYOUTS.includes(layout)
+            && !srcset
+          )
+        });
+    return (
+      await Promise.all(
+        incorrectImages.map((_, e) => {
+          const s = $(e).toString();
+          return this.warn(
+            `[${s}] should define a srcset. Using AMP Optimizer might help.`
```

Comment:
Sorry, I should have clarified -- this looks like it is emitting a warning for every image that is missing a srcset. It may be more manageable to send a single warning that lists all the images that are missing a srcset, similar to the images missing alt text:
https://github.com/ampproject/amp-toolbox/blob/25a4df393e170938e0fa982216fbe8a0fb71591a/packages/linter/src/rules/ImagesHaveAltText.ts#L24-L32

Reply:
I'm open to emitting multiple warnings if that sounds like a better approach to you, what do you think?

Reply:
There are all possible approaches in the different linter tests, so there is no "right" way to do this.
Some checks emit one warning without details to the findings, some emit one warning with details to the findings like you suggested and some emit a warning for each finding.

If we ever want to show the results in the frontend I think a list of warnings is a better approach. 
We might be able to add other metadata to each entry that can be used to display a nice list like in your screenshot.
With only one warning entry this will never be possible.